src/Common/tests/gtest_sieve_cache.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iomanip>
+#include <gtest/gtest.h>
+#include <Common/CacheBase.h>
+
+TEST(SIEVECache, set)
+{
+    using SimpleCacheBase = DB::CacheBase<int, int>;
+    auto sieve_cache = SimpleCacheBase("SIEVE", /*max_size_in_bytes*/ 10, /*max_count*/ 10, /*size_ratio*/ 0.5);
+    sieve_cache.set(1, std::make_shared<int>(2));
+    sieve_cache.set(2, std::make_shared<int>(3));
+
+    auto w = sieve_cache.sizeInBytes();
+    auto n = sieve_cache.count();
+    ASSERT_EQ(w, 2);
+    ASSERT_EQ(n, 2);
+}
+
+TEST(SIEVECache, update)
+{
+    using SimpleCacheBase = DB::CacheBase<int, int>;
+    auto sieve_cache = SimpleCacheBase("SIEVE", /*max_size_in_bytes*/ 10, /*max_count*/ 10, /*size_ratio*/ 0.5);
+    sieve_cache.set(1, std::make_shared<int>(2));
+    sieve_cache.set(1, std::make_shared<int>(3));
+    auto val = sieve_cache.get(1);
+    ASSERT_TRUE(val != nullptr);
+    ASSERT_TRUE(*val == 3);
+}
+
+TEST(SIEVECache, get)
+{
+    using SimpleCacheBase = DB::CacheBase<int, int>;
+    auto sieve_cache = SimpleCacheBase("SIEVE", /*max_size_in_bytes*/ 10, /*max_count*/ 10, /*size_ratio*/ 0.5);
+    sieve_cache.set(1, std::make_shared<int>(2));
+    sieve_cache.set(2, std::make_shared<int>(3));
+    SimpleCacheBase::MappedPtr value = sieve_cache.get(1);
+    ASSERT_TRUE(value != nullptr);
+    ASSERT_EQ(*value, 2);
+
+    value = sieve_cache.get(2);
+    ASSERT_TRUE(value != nullptr);
+    ASSERT_EQ(*value, 3);
+}
+
+struct ValueWeight
+{
+    size_t operator()(const size_t & x) const { return x; }
+};
+
+TEST(SIEVECache, evictOnSize)
+{
+    using SimpleCacheBase = DB::CacheBase<int, size_t>;
+    auto sieve_cache = SimpleCacheBase("SIEVE", /*max_size_in_bytes*/ 20, /*max_count*/ 3, /*size_ratio*/ 0.5);
+    sieve_cache.set(1, std::make_shared<size_t>(2));
+    sieve_cache.set(2, std::make_shared<size_t>(3));
+    sieve_cache.set(3, std::make_shared<size_t>(4));
+    sieve_cache.set(4, std::make_shared<size_t>(5));
+
+    auto n = sieve_cache.count();
+    ASSERT_EQ(n, 3);
+
+    auto value = sieve_cache.get(1);
+    ASSERT_TRUE(value == nullptr);
+}
+
+TEST(SIEVECache, evictOnWeight)
```

Comment:
Please add a more complex test which will check the SIEVE logic, e.g. let's construct a situation that we have in the FIFO queue a `visited` flag set to `1 1 1 0 1 0 1` and let's check that after first `removeOverflow` call we will get `0 0 0 1 0 1` (element on 4th position should have been removed). This can be done via etending `dump` method to include `visited` flag.

Reply:
Please do not forget to add this test (I see you keep resolving this comment, but it is required to add proper tests)